tests/e2e/parallel/main_test.go

```diff
@@ -2290,6 +2291,1146 @@ func TestScheduledSuiteUpdate(t *testing.T) {
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ssbName,
+			Namespace: testNamespace,
+		},
+		Profiles: []compv1alpha1.NamedObjectReference{
+			{
+				APIGroup: "compliance.openshift.io/v1alpha1",
+				Kind:     "TailoredProfile",
+				Name:     tpName,
+			},
+		},
+		SettingsRef: &compv1alpha1.NamedObjectReference{
+			APIGroup: "compliance.openshift.io/v1alpha1",
+			Kind:     "ScanSetting",
+			Name:     "default",
+		},
+	}
+
+	err = f.Client.Create(context.TODO(), ssb, nil)
+	if err != nil {
+		t.Fatalf("Failed to create ScanSettingBinding: %v", err)
+	}
+	defer f.Client.Delete(context.TODO(), ssb)
+
+	// Wait for ScanSettingBinding to become ready
+	err = f.WaitForScanSettingBindingStatus(testNamespace, ssbName, compv1alpha1.ScanSettingBindingPhaseReady)
+	if err != nil {
+		t.Fatalf("Failed waiting for ScanSettingBinding to become ready: %v", err)
+	}
+	t.Logf("ScanSettingBinding %s is now ready", ssbName)
+
+	// Wait for suite to be created and for scans to complete
+	suiteName := ssbName
+
+	// Wait for scans to complete
+	err = f.WaitForSuiteScansStatus(testNamespace, suiteName, compv1alpha1.PhaseDone, compv1alpha1.ResultNonCompliant)
+	if err != nil {
+		t.Fatalf("Failed waiting for suite scans to complete: %v", err)
+	}
+
+	t.Log("CustomRule with multiple inputs test completed successfully.")
+}
+
+func TestCustomRuleValidation(t *testing.T) {
+	t.Parallel()
+	f := framework.Global
+
+	testName := framework.GetObjNameFromTest(t)
+	testNamespace := f.OperatorNamespace
+
+	// Test 1: Invalid CEL expression
+	invalidRule := &compv1alpha1.CustomRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-invalid", testName),
+			Namespace: testNamespace,
+		},
+		Spec: compv1alpha1.CustomRuleSpec{
+			RulePayload: compv1alpha1.RulePayload{
+				ID:          fmt.Sprintf("%s-invalid", testName),
+				Title:       "Invalid Rule",
+				Description: "This rule has invalid CEL expression",
+				Severity:    "low",
+			},
+			CustomRulePayload: compv1alpha1.CustomRulePayload{
+				ScannerType: compv1alpha1.ScannerTypeCEL,
+				Expression: `
+					pods.items.all(pod,
+						invalid_function_that_doesnt_exist(pod)
+					)
+				`,
+				Inputs: []compv1alpha1.InputPayload{
+					{
+						Name: "pods",
+						KubernetesInputSpec: compv1alpha1.KubernetesInputSpec{
+							APIVersion: "v1",
+							Resource:   "pods",
+						},
+					},
+				},
+				FailureReason: "This should fail validation",
+			},
+		},
+	}
+
+	err := f.Client.Create(context.TODO(), invalidRule, nil)
+	if err != nil {
+		t.Fatalf("Failed to create CustomRule: %v", err)
+	}
+	defer f.Client.Delete(context.TODO(), invalidRule)
+
+	// Wait and expect the rule to have Error status
+	err = f.WaitForCustomRuleStatus(testNamespace, fmt.Sprintf("%s-invalid", testName), "Error")
+	if err != nil {
+		t.Fatalf("CustomRule validation failed: %v", err)
+	}
+	t.Log("CustomRule validation correctly rejected invalid expression")
+
+	// Test 2: Rule with undeclared variable
```

Comment:
Minor refactoring we can do in the future - but it can help the readability of the tests to have each of these test cases isolated into their own test case, instead of grouping them all in a single test.

Something we can work on in the future if we decide we want to organize it that way.